syntax = "proto3";

service KeyValueService {
  rpc GetValue(GetValueRequest) returns (GetValueResponse) {}
  rpc StoreValue(StoreValueRequest) returns (StoreValueResponse) {}
}

message StoreValueResponse {
  int32 code = 1;
  string message = 2;
}

message GetValueRequest {
  string key = 1;
}

message GetValueResponse {
  string value = 1;
}

message StoreValueRequest {
  string key = 1;
  string value = 2;
  repeated int32 spreading_request_prefix = 3;
}
